packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/climatedata/ClimateDataController.java

```diff
@@ -2,70 +2,183 @@
 
 
 import com.fasterxml.jackson.databind.JsonNode;
-import org.springframework.beans.factory.annotation.Autowired;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import feign.FeignException;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.security.access.annotation.Secured;
 import org.springframework.web.bind.annotation.*;
-import software.uncharted.terarium.hmiserver.models.funman.FunmanPostQueriesRequest;
+import org.springframework.web.server.ResponseStatusException;
+import software.uncharted.terarium.hmiserver.models.climateData.ClimateDataResponse;
+import software.uncharted.terarium.hmiserver.models.dataservice.dataset.Dataset;
 import software.uncharted.terarium.hmiserver.proxies.climatedata.ClimateDataProxy;
-import software.uncharted.terarium.hmiserver.proxies.funman.FunmanProxy;
 import software.uncharted.terarium.hmiserver.security.Roles;
+import software.uncharted.terarium.hmiserver.service.climatedata.ClimateDataService;
 
-import java.util.UUID;
+import java.util.ArrayList;
+import java.util.List;
 
 @RestController
 @RequestMapping("/climatedata/queries")
+@RequiredArgsConstructor
+@Slf4j
 public class ClimateDataController {
 
-	@Autowired
-	private ClimateDataProxy climateDataProxy;
+	private final ObjectMapper objectMapper;
 
-	@GetMapping("/status/{uuid}")
-	@Secured(Roles.USER)
-	public ResponseEntity<JsonNode> status(@PathVariable UUID uuid) {
-		ResponseEntity<JsonNode> response = climateDataProxy.status(uuid.toString());
+	private final ClimateDataProxy climateDataProxy;
 
-		return ResponseEntity.ok(response.getBody());
-	}
+	private final ClimateDataService climateDataService;
 
 	@GetMapping("/search-esgf")
 	@Secured(Roles.USER)
-	public ResponseEntity<JsonNode> searchEsgf(@RequestParam("query") String query) {
-		ResponseEntity<JsonNode> response = climateDataProxy.searchEsgf(query);
+	@ApiResponses(value = {
+			@ApiResponse(responseCode = "200", description = "Search ESGF and get a list of datasets"),
+			@ApiResponse(responseCode = "500", description = "Internal server error")
+	})
+	public ResponseEntity<List<Dataset>> searchEsgf(@RequestParam("query") final String query) {
+		try {
+			final ResponseEntity<JsonNode> response = climateDataProxy.searchEsgf(query);
+			if(response == null || response.getBody() == null){
+				throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, "Search ESGF failed.");
```

Comment:
if we throw here, that gets eaten up by the catch below. Can we gather information from the response and put in a regular exception?

Reply:
Thanks! now catching and rethrowing responsestatusexception, and taking care to save original status code.